utils/warnings.ml

```diff
@@ -970,12 +970,14 @@ let message = function
         let vars = List.sort String.compare vars in
         match vars with
         | [] -> assert false
-        | [x] -> "variable " ^ x
+        | [x] -> "variable " ^ x ^ " appears"
         | _::_ ->
-            "variables " ^ String.concat "," vars in
+            "variables " ^ String.concat "," vars ^ " appear" in
       Printf.sprintf
         "Ambiguous or-pattern variables under guard;\n\
-         %s may match different arguments. %t"
+         %s in different places in different or-pattern alternatives.\n\
```

Comment:
I'm not fond of the new code, which I find harder to understand due to the treatment of `appear[s]`: `%s may match [...]` reads well grammatically, but `%s in different places` does not.

Suggestion:

```ocaml
let msg =
  let vars = List.sort String.compare vars in
  let in_different_places = "in different places in different or-pattern alternatives." in
  match vars with
  | [...]
  | [x] -> "variable " ^ x ^ "appear " ^ in_different_places
  | [...]
in
[...]
```